core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -557,11 +564,18 @@ private static MetadataUpdate readAddSnapshot(JsonNode node) {
   private static MetadataUpdate readRemoveSnapshots(JsonNode node) {
     Set<Long> snapshotIds = JsonUtil.getLongSetOrNull(SNAPSHOT_IDS, node);
     Preconditions.checkArgument(
-        snapshotIds != null && snapshotIds.size() == 1,
-        "Invalid set of snapshot ids to remove. Expected one value but received: %s",
+        snapshotIds != null,
+        "Invalid set of snapshot ids to remove. Expected at least 1 value but received: %s",
```

Comment:
`snapshotIds` will be null in this case, so no need to include the variable in the error msg:
```suggestion
        "Invalid set of snapshot ids to remove: must be non-null",
```